### PR TITLE
Add mutation testing to common workspace

### PR DESCRIPTION
**Pull request by jamesobrooks**
# Purpose

Introduce mutation testing to the \`common\` workspace and systematically close mutation gaps across ten files, raising the overall mutation score from 72.67% to approximately 83%.

Mutation testing surfaces test gaps that line/branch coverage cannot — specifically, tests that pass even when logic is subtly broken. Line coverage of 80%+ does not mean the tests would catch a bug; mutation testing does.

# Major Changes

Added Stryker mutation testing (\`@stryker-mutator/core\`, \`vitest-runner\`, \`typescript-checker\`) to the \`common\` workspace with \`perTest\` coverage analysis and a \`stryker.config.mjs\` targeting all source files.

Identified and closed mutation gaps across ten files:

| File | Before | After | Notes |
|------|--------|-------|-------|
| session.ts | 0% | 100% | Added conditional branch tests for optional email field |
| auditable.ts | 0% | 100% | Replaced circular constant-import assertions with literals |
| feature-flags.ts | 0% | 100% | Replaced brittle snapshot with behavioral every(v===true) |
| actions.ts | 50% | 100% | Split bundled tests; added method/immutability assertions |
| trustees-validators.ts | 68.57% | 100% | Added ArrayDeclaration and full-name formula tests |
| search-validators.ts | 72.85% | 90.73% | Added criteria coverage; 9 equivalent mutants remain |
| phone-helper.ts | 76.27% | 88.14% | Removed anti-pattern assertions; 7 equivalent mutants remain |
| trustee-upcoming-key-dates.ts | 85.98% | 91.67% | Deduplicated factories; merged scattered describe blocks |
| sanitization.ts | 65.36% | 80.39% | Added deep-core callback/depth/circular-ref tests; 29 equivalent mutants remain |
| regex.ts | 64.66% | 99.25% | Added ~93 URL/protocol tests; 1 equivalent mutant remains |

Overall: 72.67% → ~83% (974 tests passing, no regressions)

# Testing/Validation

Mutation testing was run with Stryker after each file was addressed. All 974 tests pass. The remaining surviving mutants are documented equivalent mutants — mutations that produce identical observable behavior — not killable test gaps.

# Key Findings

Circular constant assertions: When a test imports and asserts against the same constant being mutated, Stryker mutates both sides identically so the test still passes. Example: auditable.ts tests asserted \`result.lastModifiedBy === SYSTEM_USER_REFERENCE\` — when Stryker changed SYSTEM_USER_REFERENCE.id to \`''\`, the expected value changed too. Fix: assert on literal strings.

Behavioral vs. implementation tests: Tests that check exact boundary values ("must be exactly 15") rather than behavioral outcomes ("rejects 16, accepts 15") fail to kill off-by-one mutants. The fix was not to add more tests but to write tests at the right level of abstraction.

Equivalent mutants: Some mutations cannot be killed because they produce identical behavior. The dominant pattern in this codebase is sanitizeUrl pre-filter conditions: conditions that guard against values the downstream regex already rejects. Adding tests for those inputs does not kill the mutant — no test can.

Anti-Stryker test artifacts removed: Several tests contained \`expect(result).not.toEqual('Stryker was here!')\` assertions and inline comments like "// Kills mutant: ...". These are meaningless in a real test suite and were removed.

# Notes

Run mutation tests locally: \`npx stryker run --config common/stryker.config.mjs\` from the repo root, or \`npm run test:mutation\` from the \`common\` directory.

# Definition of Done:

- [ ] Code refactored for clarity: Developers can understand the work simply by reviewing the code
- [ ] Dependency rule followed: More important code doesn't directly depend on less important code
- [ ] Development debt eliminated: UX and code aligns to the team's latest understanding of the domain
- [ ] No regressions: Changes do not cause regression in related or unrelated areas of the application

### CODE REVIEW

**Review by jalba-flexion — APPROVED** [review]

**Review by mattstankey — APPROVED** [review]

File: common/stryker.config.mjs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  coverageAnalysis: 'perTest',
+  reporters: ['html', 'clear-text', 'progress'],
+  htmlReporter: {
+    fileName: 'reports/mutation/mutation.html',
+  },
+  thresholds: {
+    high: 80,
+    low: 60,
+    break: null,
+  },
```
[the remark this answers is not in the document]
Comment (jamesobrooks) [reply common/stryker.config.mjs:21]: We're not automating as of now, so the pipeline never even runs it.
